src/builder.rs

```diff
@@ -0,0 +1,177 @@
+//! Builder types to compose layers and services
+
+use futures::{Async, Future, Poll};
+use std::fmt;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use tower_layer::{util::Chain, Layer, LayerExt};
+use tower_service::Service;
+
+/// Configure and build a `MakeService`
+///
+/// `ServiceBuilder` collects layers and a `MakeService` transport
+/// and produces a new `MakeService` that is wrapped by the composed
+/// layer.
+#[derive(Debug)]
+pub struct ServiceBuilder<S, L, Target, Request>
+where
+    S: Service<Target>,
```

Comment:
I don't think you need these bounds at the struct level. You should be able to get away without any bounds or phantom data.

Reply:
so I tried this but I thought I needed them because of this
![image](https://user-images.githubusercontent.com/5758045/54404133-ed3a9400-46a8-11e9-902c-97dc249e4dfa.png)